Source/LanderLads/SpaceshipPawn.cpp

```diff
@@ -0,0 +1,226 @@
+{
+	constexpr float RelativeXThrust = 490 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ?
+		(this->GetWorld()->GetRealTimeSeconds() - this->ForwardThrustStartTime) :
+		(this->GetWorld()->GetRealTimeSeconds() - this->BackwardThrustStartTime);
+	
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float ForwardBackwardVectorMagnitude = RelativeXThrust * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::X, ForwardBackwardVectorMagnitude);
+}
+
+void ASpaceshipPawn::MoveLeftRight(float AxisValue)
+{
+	constexpr float RelativeYThrust = 490 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ?
+		(this->GetWorld()->GetRealTimeSeconds() - this->RightThrustStartTime) :
+		(this->GetWorld()->GetRealTimeSeconds() - this->LeftThrustStartTime);
+
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float ForwardBackwardVectorMagnitude = RelativeYThrust * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::Y, ForwardBackwardVectorMagnitude);
+}
+
+void ASpaceshipPawn::MoveUpDown(float AxisValue)
+{
+	const float Lift = this->GetWorld()->GetGravityZ() * -1 * 4000;
+	const float ClampedAxisValue = ClampAxisValue(AxisValue);
+
+	const float TimeDifference = ClampedAxisValue > 0.0f ? 
+		(this->GetWorld()->GetRealTimeSeconds() - this->UpThrustStartTime) : 
+		(this->GetWorld()->GetRealTimeSeconds() - this->DownThrustStartTime);
+
+	const float Curvey = this->CurveFloat->GetFloatValue(TimeDifference);
+	const float UpDownForceVectorMagnitude = Lift * ClampedAxisValue * Curvey;
+	this->ResultantForceVector.SetComponentForAxis(EAxis::Z, UpDownForceVectorMagnitude);
+	
+}
+
+void ASpaceshipPawn::MoveCameraLookHorizontal(float AxisValue)
+{
+	float ClampedAxisValue = ClampAxisValue(AxisValue);
+	this->AddControllerYawInput(
+		this->GetWorld()->GetDeltaSeconds() * 45 * ClampedAxisValue);
+}
+
+void ASpaceshipPawn::MoveCameraLookVertical(float AxisValue)
+{
+	float ClampedAxisValue = ClampAxisValue(AxisValue);
+	this->AddControllerPitchInput(
+		this->GetWorld()->GetDeltaSeconds() * -45 * ClampedAxisValue);
+}
+
+const float ASpaceshipPawn::ClampAxisValue(float AxisValue)
+{
+	return FMath::Clamp(AxisValue, -1.0f, 1.0f);
+}
+
+USpaceshipStaticMeshComponent* ASpaceshipPawn::GetSpaceshipStaticMeshComponent()
+{
+	return this->SpaceshipStaticMeshComponent;
+}
+
+void ASpaceshipPawn::OnCompHit(
+	UPrimitiveComponent* HitComp,
+	AActor* OtherActor,
+	UPrimitiveComponent* OtherComp,
+	FVector NormalImpulse,
+	const FHitResult& Hit)
+{
+	if (HitComp == this->GetSpaceshipStaticMeshComponent()) {
+		constexpr float ForceThreshold = 300000;
+		if (NormalImpulse.X > ForceThreshold ||
+			NormalImpulse.Y > ForceThreshold ||
+			NormalImpulse.Z > ForceThreshold) {
+			this->ExplodeShip();
```

Comment:
If you only use this once, just put the function in here so we don't pollute the header file.